src/telliot_feeds/integrations/diva_protocol/report.py

```diff
@@ -36,20 +36,40 @@ class DIVAProtocolReporter(TellorFlexReporter):
     DIVA Protocol Reporter
     """
 
-    def __init__(self, *args, **kwargs) -> None:  # type: ignore
+    def __init__(  # type: ignore
+        self,
+        middleware_address: str = "0xF3F62041113c92F080E88200481dFE392369d17b",
+        diva_diamond_address: str = "0x27D1BD739BD152CDaE38d4444E9aee3498166f01",
+        network_name: str = "goerli",
+        extra_undisputed_time: int = 0,
+        wait_before_settle: int = 0,
+        *args,
+        **kwargs,
+    ) -> None:
         super().__init__(*args, **kwargs)
+        self.extra_undisputed_time = extra_undisputed_time
+        self.wait_before_settle = wait_before_settle
         self.settle_period: Optional[int] = None
+        self.network_name = network_name
+        self.diva_diamond_address = diva_diamond_address
         self.middleware_contract = DivaOracleTellorContract(self.endpoint, self.account)
+        self.middleware_contract.address = middleware_address
         self.middleware_contract.connect()
 
     async def filter_unreported_pools(self, pools: list[DivaPool]) -> list[DivaPool]:
         """
         Retrieves first unreported pool.
         """
+        # also check against local cache of reported pools
+        local_stored_reported = get_reported_pools()
         unreported_pools = []
         for pool in pools:
+            if pool.pool_id in local_stored_reported:
+                logger.info(f"Pool {pool.pool_id} already reported. Checked against local storage.")
+                continue
+
             query = DIVAProtocol(
-                poolId=pool.pool_id, divaDiamond="0x2d8642777C51dB31945CeDbbC3198d75e497cb48", chainId=5
+                poolId=pool.pool_id, divaDiamond=self.diva_diamond_address, chainId=self.endpoint.chain_id
```

Comment:
I don't think there need to be handling here since it's an instance of the diva query type, and the params it takes are never going to be None I think